### PR TITLE
tests: ensure samba-metrics sidecar is responsive

### PR DESCRIPTION
**Pull request by synarete**
Issue a curl from samba container to samba-metrics sidecar container in
order to verify samba-metrics is responsive with valid output. Execute
the curl command within the pod itself.

Enabled only if 'samba-metrics' container exists in pod.

Signed-off-by: Shachar Sharon <ssharon@redhat.com>

### CODE REVIEW

File: tests/integration/smb_share_test.go [hunk 1]
```diff
@@ -273,6 +282,57 @@ func (s *SmbShareWithExternalNetSuite) TestServiceIsLoadBalancer() {
 	)
 }
 
+func (s *SmbShareSuite) TestMetricsOnPod() {
+	pod, cont, err := s.getMetricsContainer()
+	s.Require().NoError(err)
+	if cont == nil {
+		return // OK, running without metrics; ignored
```
Comment (phlogistonjohn) [inline tests/integration/smb_share_test.go:289]: Please change this to a test skip. Example: `s.T().Skipf("no metrics container present")`

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Body of the test looks OK to me. I'd just like to use an explicit test skip to distinguish from the success case when the test actually runs.

File: tests/integration/smb_share_test.go (hunk 1 above)
Comment (synarete) [reply tests/integration/smb_share_test.go]: ACK

**Comment by synarete** [comment]
@anoopcs9 @phlogistonjohn CI testing keep on failing on this PR as well as my other (trivial) PR#196. I  re-checked both on my setup (with minikube) and PASS with not errors. Any idea what could cause those failures?

**Comment by anoopcs9** [comment]
> @anoopcs9 @phlogistonjohn CI testing keep on failing on this PR as well as my other (trivial) PR#196. I re-checked both on my setup (with minikube) and PASS with not errors. Any idea what could cause those failures?

I'm looking into CentOS CI test failures..I'm able to reproduce where I am seeing following errors from Ceph cluster which prevents it to come up HEALTHY:

`MON_CLOCK_SKEW`
`MON_DISK_LOW`

clcok skew is surprising but I think I figured out a way to get around `MON_DISK_LOW` error.

**Comment by anoopcs9** [comment]
> > @anoopcs9 @phlogistonjohn CI testing keep on failing on this PR as well as my other (trivial) PR#196. I re-checked both on my setup (with minikube) and PASS with not errors. Any idea what could cause those failures?
> 
> I'm looking into CentOS CI test failures..I'm able to reproduce where I am seeing following errors from Ceph cluster which prevents it to come up HEALTHY:
> 
> `MON_CLOCK_SKEW` `MON_DISK_LOW`
> 
> clock skew is surprising but I think I figured out a way to get around `MON_DISK_LOW` error.

#197 should fix rook-ceph errors from CentOS CI test runs.

**Comment by synarete** [comment]
> > > @anoopcs9 @phlogistonjohn CI testing keep on failing on this PR as well as my other (trivial) PR#196. I re-checked both on my setup (with minikube) and PASS with not errors. Any idea what could cause those failures?
> > 
> > 
> > I'm looking into CentOS CI test failures..I'm able to reproduce where I am seeing following errors from Ceph cluster which prevents it to come up HEALTHY:
> > `MON_CLOCK_SKEW` `MON_DISK_LOW`
> > clock skew is surprising but I think I figured out a way to get around `MON_DISK_LOW` error.
> 
> #197 should fix rook-ceph errors from CentOS CI test runs.

Cool! many thanks @anoopcs9

**Review by phlogistonjohn — APPROVED** [review]

**Pull request merged**